pkg/chartutil/create.go

```diff
@@ -435,7 +435,7 @@ app.kubernetes.io/managed-by: {{ .Release.Service }}
 Selector labels
 */}}
 {{- define "<CHARTNAME>.selectorLabels" -}}
-app.kubernetes.io/name: {{ include "<CHARTNAME>.name" . }}
+app.kubernetes.io/name: {{ include "<CHARTNAME>.name" . | toJson }}
```

Comment:
Why is the name being piped to `toJson`?

Reply:
The `name: %q` above fixes this.

```console
$ helm create no                                                                                                                                                                                     
Creating no

$ helm template ./no | grep 'false'
# Source: false/templates/serviceaccount.yaml
  name: RELEASE-NAME-false
    helm.sh/chart: false-0.1.0
    app.kubernetes.io/name: false
# Source: false/templates/service.yaml
  name: RELEASE-NAME-false
    helm.sh/chart: false-0.1.0
    app.kubernetes.io/name: false
    app.kubernetes.io/name: false
# Source: false/templates/deployment.yaml
  name: RELEASE-NAME-false
    helm.sh/chart: false-0.1.0
    app.kubernetes.io/name: false
      app.kubernetes.io/name: false
        app.kubernetes.io/name: false
      serviceAccountName: RELEASE-NAME-false
        - name: false
# Source: false/templates/tests/test-connection.yaml
  name: "RELEASE-NAME-false-test-connection"
    helm.sh/chart: false-0.1.0
    app.kubernetes.io/name: false
      args: ['RELEASE-NAME-false:80']
```

This one, fixes that.

```
helm template ./no | grep -P '\bno\b'
# Source: no/templates/serviceaccount.yaml
  name: RELEASE-NAME-no
    helm.sh/chart: no-0.1.0
    app.kubernetes.io/name: no
# Source: no/templates/service.yaml
  name: RELEASE-NAME-no
    helm.sh/chart: no-0.1.0
    app.kubernetes.io/name: no
    app.kubernetes.io/name: no
# Source: no/templates/deployment.yaml
  name: RELEASE-NAME-no
    helm.sh/chart: no-0.1.0
    app.kubernetes.io/name: no
      app.kubernetes.io/name: no
        app.kubernetes.io/name: no
      serviceAccountName: RELEASE-NAME-no
        - name: no
# Source: no/templates/tests/test-connection.yaml
  name: "RELEASE-NAME-no-test-connection"
    helm.sh/chart: no-0.1.0
    app.kubernetes.io/name: no
      args: ['RELEASE-NAME-no:80']
```

Reply:
I think that instead of `| toJson`, you want `| quote`